mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1076,10 +1109,13 @@ struct GridwiseGemmV2RewritePattern
     // Prepare some useful constants.
     Value zeroConstantOp = b.create<ConstantIndexOp>(loc, 0);
 
+    Value matA = unKpack(b, op.a());
+    Value matB = unKpack(b, op.b());
+
     // Obtain critical matrix dimensions.
     ArrayRef<int64_t> aShape, bShape, cShape;
-    aShape = op.a().getType().template cast<MemRefType>().getShape();
-    bShape = op.b().getType().template cast<MemRefType>().getShape();
+    aShape = matA.getType().template cast<MemRefType>().getShape();
```

Comment:
Is this safe? Where else do we use aShape?

Reply:
Addressed in 2bbf5c6. It's not... I was looking at `aShape` and bShape` only previously and you reminded me that those will determine the `G, K, M, N` sizes which will be different without kpack. Reverted back to op.a and op.b and wired thing about dimension typo caused regression also went away.